persistence/sql/identity/persister_identity.go

```diff
@@ -666,56 +666,91 @@ func (p *IdentityPersister) ListIdentities(ctx context.Context, params identity.
 
 	con := p.GetConnection(ctx)
 	nid := p.NetworkID(ctx)
-	query := con.Where("identities.nid = ?", nid).Order("identities.id DESC")
 
-	// Credentials are not expanded through `EagerPreload` but manually after
-	// fetching the identities, hence we filter out the relevant expand options.
-	var expandExceptCredentials sqlxx.Expandables
-	for _, e := range params.Expand {
-		if e != identity.ExpandFieldCredentials {
-			expandExceptCredentials = append(expandExceptCredentials, e)
+	joins := ""
+	wheres := ""
+	args := []any{nid}
+	identifier := params.CredentialsIdentifier
+	identifierOperator := "="
+	if identifier == "" && params.CredentialsIdentifierSimilar != "" {
+		identifier = params.CredentialsIdentifierSimilar
+		identifierOperator = "%"
+		switch con.Dialect.Name() {
+		case "postgres", "cockroach":
+		default:
+			identifier = "%" + identifier + "%"
+			identifierOperator = "LIKE"
```

Comment:
This is the mysql and sqlite case. Mysql does support full-text and fuzzy search, and special index types to allow them. However, it was not easy to understand at all, and I found no easy way to have the same behavior as postgres/CRDB. I think this is already an improvement on the current situation, and this feature is open for contributions!